diff --git a/node_modules/@headlessui/react/dist/headlessui.dev.cjs b/node_modules/@headlessui/react/dist/headlessui.dev.cjs
index 2c5794f..27b2d73 100644
--- a/node_modules/@headlessui/react/dist/headlessui.dev.cjs
+++ b/node_modules/@headlessui/react/dist/headlessui.dev.cjs
@@ -443,10 +443,13 @@ function $507fabe10e71c6fb$var$setupGlobalFocusEvents(element) {
   const windowObject = (0, $431fbd86ca7dc216$export$f21a1ffae260145a)(element);
   const documentObject = (0, $431fbd86ca7dc216$export$b204af158042fbac)(element);
   let focus = windowObject.HTMLElement.prototype.focus;
-  windowObject.HTMLElement.prototype.focus = function() {
-    $507fabe10e71c6fb$var$hasEventBeforeFocus = true;
-    focus.apply(this, arguments);
-  };
+  try {
+    windowObject.HTMLElement.prototype.focus = function() {
+      $507fabe10e71c6fb$var$hasEventBeforeFocus = true;
+      focus.apply(this, arguments);
+    };
+  }
+  catch {}
   documentObject.addEventListener("keydown", $507fabe10e71c6fb$var$handleKeyboardEvent, true);
   documentObject.addEventListener("keyup", $507fabe10e71c6fb$var$handleKeyboardEvent, true);
   documentObject.addEventListener("click", $507fabe10e71c6fb$var$handleClickEvent, true);
@@ -477,7 +480,11 @@ var $507fabe10e71c6fb$var$tearDownWindowFocusTracking = (element, loadListener)
     documentObject.removeEventListener("DOMContentLoaded", loadListener);
   if (!$507fabe10e71c6fb$export$d90243b58daecda7.has(windowObject))
     return;
-  windowObject.HTMLElement.prototype.focus = $507fabe10e71c6fb$export$d90243b58daecda7.get(windowObject).focus;
+  // Firefox CSP Guard
+  try {
+    windowObject.HTMLElement.prototype.focus = $507fabe10e71c6fb$export$d90243b58daecda7.get(windowObject).focus;
+  }
+  catch {}
   documentObject.removeEventListener("keydown", $507fabe10e71c6fb$var$handleKeyboardEvent, true);
   documentObject.removeEventListener("keyup", $507fabe10e71c6fb$var$handleKeyboardEvent, true);
   documentObject.removeEventListener("click", $507fabe10e71c6fb$var$handleClickEvent, true);
